packages/riverpod/test/framework/auto_dispose_test.dart

```diff
@@ -529,6 +529,33 @@ final alwaysAlive = Provider((ref) {
     expect(container.getAllProviderElements(), isEmpty);
   });
 
+  test(
+      'when cancel autoDispose family override it should be disposed immediately',
+      () async {
+    final provider = Provider.autoDispose.family<int, int>((ref, _) => -1);
+
+    var constructionCount = 0;
+    final override = Provider.autoDispose.family<int, int>((ref, _) {
+      return ++constructionCount;
+    });
+
+    final overriden = provider.overrideWithProvider(override);
```

Comment:
Do not create variables for these :)

Reply:
In fact you could switch to overrideWith too

Reply:
I forgot to remove them after inspecting. I'll close conversations that you've fixed all.